controllers/onload_controller_test.go

```diff
@@ -0,0 +1,132 @@
+package controllers
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	onloadv1alpha1 "github.com/Xilinx-CNS/kubernetes-onload/api/v1alpha1"
+
+	kmm "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+)
+
+var _ = Describe("Testing createModule function", func() {
+	var onload *onloadv1alpha1.Onload
+
+	BeforeEach(func() {
+		onload = &onloadv1alpha1.Onload{}
+	})
+
+	It("Should work with a valid onload CR", func() {
+		_, err := createModule(onload, "example", "example")
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("Should fail if a build parameter was supplied", func() {
+		onload.Spec = onloadv1alpha1.Spec{
+			Onload: onloadv1alpha1.OnloadSpec{
+				KernelMappings: []onloadv1alpha1.OnloadKernelMapping{
+					{
+						Regexp:            "example",
+						KernelModuleImage: "example",
+						Build:             &onloadv1alpha1.OnloadKernelBuildSpec{},
+					},
+				},
+			},
+		}
+		_, err := createModule(onload, "example", "example")
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("Should have the correct number of kernel mappings", func() {
+		for i := 0; i < 10; i++ {
+			onload.Spec.Onload.KernelMappings = append(
+				onload.Spec.Onload.KernelMappings,
+				onloadv1alpha1.OnloadKernelMapping{},
+			)
+		}
+		module, err := createModule(onload, "example", "example")
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(len(module.Spec.ModuleLoader.Container.KernelMappings)).
+			To(Equal(len(onload.Spec.Onload.KernelMappings)))
+	})
+})
+
+var _ = Describe("onload controller", func() {
+	Context("testing onload controller", func() {
+		var onload *onloadv1alpha1.Onload
+		var testNamespace *corev1.Namespace
+
+		BeforeEach(func() {
+			namespaceName := "test-ns-" + rand.String(5)
```

Comment:
The purpose of using rand here is so that each test can be given a unique namespace to operate in.
Unique namespaces are basically required because envtest doesn't have the appropriate controller that handles garbage collection / deletion, so everything not explicitly deleted will linger after a test which can cause issues in subsequent tests.
KMM doesn't have this issue since they don't use Envtest.

I'll add a call to `rand.Seed()` in `BeforeSuite`, and do a check to see if the namespace already exists before continuing.